eo-maven-plugin/src/main/java/org/eolang/maven/ProbeMojo.java

```diff
@@ -94,28 +100,41 @@ public final class ProbeMojo extends SafeMojo {
     @Override
     public void exec() throws IOException {
         if (this.hsh == null) {
-            this.hsh = new ChRemote(this.tag);
+            this.hsh = new ChCached(
+                new ChNarrow(
+                    new ChRemote(this.tag)
+                )
+            );
         }
         final Collection<String> probed = new HashSet<>(1);
         final Collection<ForeignTojo> tojos = this.scopedTojos().unprobed();
         for (final ForeignTojo tojo : tojos) {
             final Path src = tojo.optimized();
-            final Collection<String> names = this.probes(src);
-            if (!names.isEmpty()) {
-                Logger.info(this, "Probing object(s): %s", names);
+            final Collection<ObjectName> objects = this.probes(src);
+            if (!objects.isEmpty()) {
+                Logger.info(this, "Probing object(s): %s", objects);
             }
             int count = 0;
-            for (final String name : names) {
-                if (!this.objectionaries.contains(this.hsh, name)) {
+            for (final ObjectName object : objects) {
+                if (!this.objectionaries.contains(object.hash(), object.value())) {
                     continue;
                 }
                 ++count;
                 this.scopedTojos()
-                    .add(name)
+                    .add(object.asString())
```

Comment:
@maxonfjvipon Maybe it's better to pass the `object` right to `add()` method instead of `string`?

Something like:
```java
this.scopedTojos().add(object);
```
The same for `probed.add(...)`